### PR TITLE
Olwa/dev

### PR DESCRIPTION
**Pull request by owalk**
I fixed the collision logic!!! Super excited about this one :)

### CODE REVIEW

File: game/entities/entity.py [hunk 1]
```diff
@@ -391,6 +410,151 @@ def facing_towards_entity(self, other_sprite: pygame.sprite) -> bool:
...
+                x_dist_out_hitbox = collision_rect.left - self.rect.right
+                self.rect.move_ip(x_dist_out_hitbox, 0)
+            # collided sprite is on the left
+            else:
+                x_dist_out_hitbox = collision_rect.right - self.rect.left
+                self.rect.move_ip(x_dist_out_hitbox, 0)
+        else:
+            # top or bottom collision
+            if self.hitbox.centery < collision_rect.centery:
+                # teleport to the left
```
Comment (Sean-Nishi) [inline game/entities/entity.py:501]: Comment should be teleport to up or down

**Review by Sean-Nishi — CHANGES_REQUESTED** [review]
All good except for the one comment not describing the right direction

File: game/entities/entity.py (hunk 1 above)
Comment (owalk) [reply game/entities/entity.py]: I fixed a couple of other comments in this method as well as the one you pointed out. Good catch

**Comment by owalk** [comment]
I found two additional bugs this morning. The self colliding needs a little wiggle room when being removed from inside a collision rect. fixed below:
image
Additionally, vector reflections were able to get stuck going back and forth even with this change, so I added additional checks for this previously uncaught edge case
image

**Review by Sean-Nishi — APPROVED** [review]
Good job removing collision_direction in player's collision_set_compass. Using math vector2s is so much cleaner

**Pull request merged**